simexpal/queuesock.py

```diff
@@ -50,6 +63,13 @@ def run(self):
 					connection = Connection(conn)
 
 					self.selector.register(conn, selectors.EVENT_READ, connection)
+				elif sk.fileobj == signal_reader:
+					if cur_subproc is not None:
+						cur_subproc.terminate()
+					os.close(signal_reader)
+					os.close(signal_writer)
+					self.close()
+					sys.exit(1)
```

Comment:
As in common.py, the usual event loop logic should handle this without any extra code.

Reply:
Analogously to ``common.py`` we need to terminate the parent process. I changed up the handling of this case to avoid some duplicate code